common/src/main/java/com/microsoft/identity/common/internal/ui/webview/EmbeddedWebViewAuthorizationStrategy.java

```diff
@@ -118,22 +125,25 @@ private Intent buildAuthorizationActivityStartIntent(
         // RedirectURI used to get the auth code in nested app auth is that of a hub app (brkRedirectURI)       
         final String redirectUri = mAuthorizationRequest.getBrkRedirectUri() != null ? mAuthorizationRequest.getBrkRedirectUri() : mAuthorizationRequest.getRedirectUri();
         return AuthorizationActivityFactory.getAuthorizationActivityIntent(
-                    getApplicationContext(),
-                    null,
-                    requestUrl.toString(),
-                    redirectUri,
-                    mAuthorizationRequest.getRequestHeaders(),
-                    AuthorizationAgent.WEBVIEW,
-                    mAuthorizationRequest.isWebViewZoomEnabled(),
-                    mAuthorizationRequest.isWebViewZoomControlsEnabled(),
-                    sourceLibraryName,
-                    sourceLibraryVersion);
+                getApplicationContext(),
+                getBrowserIntent(),
```

Comment:
>getBrowserIntent(), [](http://example.com/codeflow?start=16&length=19)

It appears you are re-purposing this parameter, and creating intent which could later get used if the switch browser flow is run.

Instead, can we just create intent after switch browser redirect is received. Advantage is you don't need to create intent unless switch browser flow is really started, intent can created fully after redirect (data uri + action etc) and this parameter is is not re-purposed